### ISSUE TITLE
Intention to add Checksum to detect crashes and recover?

### ISSUE THREAD

**Issue by sean-** [vadiminshakov/gowal#1]
Is there any intention to add support for checksums to the WAL so that when a crash occurs, log messages can be rolled back to a known good state (I'm thinking about something similar to how lmdb or bbold handles their crash recovery)?

**Comment by vadiminshakov** [comment]
Good point. I think this work can be planned for the near future improvements.

**Comment by vadiminshakov** [comment]
added checksums: [v0.0.1](https://github.com/vadiminshakov/gowal/releases/tag/v0.0.1)

**Comment by vadiminshakov** [comment]
added recover in [v0.0.2](https://github.com/vadiminshakov/gowal/releases/tag/v0.0.2)
